pkg/component/component.go

```diff
@@ -402,6 +340,203 @@ func getS2IPaths(podEnvs []corev1.EnvVar) []string {
+	}
+	if !versionExists {
+		log.Info("Run 'odo catalog list components' to see a list of supported component type versions")
+		return fmt.Errorf("Invalid component version %s:%s", componentType, componentVersion)
+	}
+
+	// Validate component name
+	err = validation.ValidateName(*componentSettings.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create component of name %s", *componentSettings.Name)
+	}
+
+	if isCmpExistsCheck {
+		exists, err = Exists(client, *componentSettings.Name, *componentSettings.Application)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check if component of name %s exists in application %s", *componentSettings.Name, *componentSettings.Application)
+		}
+		if exists {
+			return fmt.Errorf("component with name %s already exists in application %s", *componentSettings.Name, *componentSettings.Application)
+		}
+	}
+
+	if *componentSettings.SourceType == config.LOCAL {
+		srcLocInfo, err := os.Stat(*(componentSettings.SourceLocation))
+		if err != nil {
+			return errors.Wrap(err, "failed to create component. Please view the settings used using the command `odo config view`")
+		}
+		if !srcLocInfo.IsDir() {
+			return fmt.Errorf("source path for component created for local source needs to be a directory")
+		}
+	}
+
+	return
+}
+
+// ApplyConfig applies the component config onto component dc
+// Parameters:
+//	client: occlient instance
+//	appName: Name of application of which the component is a part
+//	componentName: Name of the component which is being patched with config
+//	componentConfig: Component configuration
+// Returns:
+//	err: Errors if any else nil
+func ApplyConfig(client *occlient.Client, componentConfig config.LocalConfigInfo, context string, stdout io.Writer) (err error) {
+
+	cmpName := componentConfig.GetName()
+	cmpSrcLoc := componentConfig.GetSourceLocation()
+	cmpSrcType := componentConfig.GetSourceType()
+
+	if cmpSrcType == config.GIT {
+		if err := Update(client, componentConfig, cmpSrcLoc, stdout); err != nil {
+			return err
+		}
+		log.Successf("The component %s was updated successfully", cmpName)
+	} else if cmpSrcType == config.LOCAL {
+		cmpSrcLoc, err := pkgUtil.GetAbsPath(cmpSrcLoc)
+		if err != nil {
+			return err
+		}
+		fileInfo, err := os.Stat(cmpSrcLoc)
+		if err != nil {
+			return err
+		}
+		if !fileInfo.IsDir() {
+			return fmt.Errorf("incorrect context passed. It should be a path to the compoennt source directory")
+		}
+		if err = Update(client, componentConfig, cmpSrcLoc, stdout); err != nil {
+			return err
+		}
+		log.Successf("The component %s was updated successfully", cmpName)
+	} else if cmpSrcType == config.BINARY {
+		cmpSrcLoc, err := pkgUtil.GetAbsPath(cmpSrcLoc)
+		if err != nil {
+			return err
+		}
+		if err = Update(client, componentConfig, cmpSrcLoc, stdout); err != nil {
+			return err
+		}
+		log.Successf("The component %s was updated successfully", cmpName)
```

Comment:
Above code needs to have documentation / be commented.

There is also no spacing... so entire code is very much cluttered together.

Reply:
Ack